builder/tools-builtin.go

```diff
@@ -52,3 +56,51 @@ func RunTool(tool string, args ...string) error {
 	}
 	return nil
 }
+
+// wasmOptLock exists because the binaryen C API uses global state.
+// In the future, we can hopefully avoid this by re-execing.
+var wasmOptLock sync.Mutex
```

Comment:
We already do this for Clang and LLD, so it wouldn't be too much trouble to add another for wasm-opt. Either way, that's easy to do at a later time (no need to do it from the beginning).